src/librustc_save_analysis/lib.rs

```diff
@@ -605,8 +606,8 @@ impl<'l, 'tcx> SaveContext<'l, 'tcx> {
         }
     }
 
-    pub fn get_path_res(&self, id: NodeId) -> Res {
-        match self.tcx.hir().get(id) {
+    pub fn get_path_res(&self, hir_id: hir::HirId) -> Res {
```

Comment:
Not sure changing this to a `HirId` is  a good idea. It seems to just add more conversion code.

Reply:
The `NodeId` eventually gets converted to a `HirId` with the call to `get` anyway; my end goal is to convert a bigger part of `librustc_save_analysis` to use `HirId` and remove the conversions while I'm at it.

Reply:
Save analysis uses the AST which means it has to use node ids. So I think you should just change this back.